components/script/dom/globalscope.rs

```diff
@@ -3211,8 +3213,26 @@ impl GlobalScope {
                                 }
                             };
 
-                            // TODO: 6. If navigator.[[hasGamepadGesture]] is false
-                            //          and gamepad contains a gamepad user gesture:
+                            if !window.Navigator().has_gamepad_gesture() {
+                                window.Navigator().set_has_gamepad_gesture(true);
+                                for i in 0..gamepad_list.Length() {
+                                    if let Some(gamepad) = gamepad_list.Item(i as u32) {
+                                        gamepad.set_exposed(true);
+                                        gamepad.update_timestamp(*current_time);
+                                        let new_gamepad = Trusted::new(&*gamepad);
+                                        if window.Document().is_fully_active() {
+                                            window.task_manager().gamepad_task_source().queue(
+                                                task!(update_gamepad_connect: move || {
+                                                    let gamepad = new_gamepad.root();
+                                                    gamepad.notify_event(GamepadEventType::Connected);
+                                                }),
+                                                window.upcast(),
+                                            )
+                                            .unwrap();
```

Comment:
Let's also use `queue_with_canceller` for all tasks(here and others), and replace `unwrap` with an appropriate `expect`.